src/nvram.cpp

```diff
@@ -0,0 +1,396 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2021 YADRO
+
+#include "edk.hpp"
+#include "nvram.hpp"
+
+#include <endian.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <cstring>
+#include <system_error>
+
+namespace nvram
+{
+
+/** @brief GUID wrapper. */
+class Guid
+{
+  public:
+    /**
+     * @brief Constructor. EFI stores GUID in the structure which is not
+     *        compatible with uuid_t (flat array).
+     *
+     * @param[in] guid GUID in EFI format
+     */
+    Guid(const EFI_GUID& guid)
+    {
+        *reinterpret_cast<uint32_t*>(&uuid[0]) = htobe32(guid.Data1);
+        *reinterpret_cast<uint16_t*>(&uuid[4]) = htobe16(guid.Data2);
+        *reinterpret_cast<uint16_t*>(&uuid[6]) = htobe16(guid.Data3);
+        memcpy(&uuid[8], &guid.Data4, sizeof(guid.Data4));
+    }
+
+    bool operator!=(const Guid& rhs) const
+    {
+        return uuid_compare(uuid, rhs.uuid) != 0;
+    }
+
+    uuid_t uuid;
+};
+
+/** @brief Map file to memory. */
+class FileMapper
+{
+  public:
+    /** @brief Destructor. */
+    ~FileMapper()
+    {
+        if (data != MAP_FAILED)
+        {
+            munmap(data, size);
+        }
+        if (fd != -1)
```

Comment:
`fd`might be closed right after `mmap()` called. So, it's not necessary to make it as a member of the class.

Reply:
Fixed